### PR TITLE
Add Slackbot.

### PR DESCRIPTION
**Pull request by nof20**
You know what Cloud Build needs?  Notifications on Slack!

### CODE REVIEW

File: slackbot/examples/cloudbuild-timeout.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+steps:
+- name: 'gcr.io/$PROJECT_ID/slackbot'
+  args: [ '--build', '$BUILD_ID',
+          '--webhook', '<Add your webhook URL here>' ]
+- name: 'gcr.io/cloud-builders/docker'
+  args: [ 'build', '.', '-f', 'Dockerfile-timeout']
+timeout: 30s
```
Comment (bendory) [inline slackbot/examples/cloudbuild-timeout.yaml]: you could make this example run faster with a timeout of 1s or 2s instead.

Comment (nof20) [reply slackbot/examples/cloudbuild-timeout.yaml]: I tried that, but you have to give it enough time to pull the Slackbot container and execute the watcher build first!

File: slackbot/main.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	webhook = flag.String("webhook", "", "Slack webhook URL")
+	mode    = flag.String("mode", "trigger", "Mode the builder runs in")
+)
+
+func main() {
+	log.Print("Starting slackbot")
+	flag.Parse()
+	ctx := context.Background()
+
+<<<<<<< HEAD
```
Comment (Philmod) [inline slackbot/main.go]: There are some merge issues to fix

Comment (nof20) [reply slackbot/main.go]: I fixed, good spot & thank you.

**Pull request merged**